website/server/controllers/api-v3/tasks.js

```diff
@@ -374,19 +408,24 @@ api.scoreTask = {
   url: '/tasks/:taskId/score/:direction',
   middlewares: [authWithHeaders()],
   async handler (req, res) {
-    req.checkParams('taskId', res.t('taskIdRequired')).notEmpty().isUUID();
     req.checkParams('direction', res.t('directionUpDown')).notEmpty().isIn(['up', 'down']);
 
     let validationErrors = req.validationErrors();
     if (validationErrors) throw validationErrors;
 
     let user = res.locals.user;
-    let direction = req.params.direction;
+    let {taskId} = req.params;
 
-    let task = await Tasks.Task.findOne({
-      _id: req.params.taskId,
-      userId: user._id,
-    }).exec();
+    let taskQuery = { userId: user._id};
+
+    if (validator.isUUID(taskId)) {
+      taskQuery._id = taskId;
+    } else {
+      taskQuery.shortName = taskId;
+    }
+
+    let task = await Tasks.Task.findOne(taskQuery).exec();
+    let direction = req.params.direction;
```

Comment:
since you're destructuring req.params at line 417 it makes sense to also define direction there?